test/standalone/test_rpc_proxy.py

```diff
@@ -249,23 +252,25 @@ def method(self, arg):
         # make an async call that will block,
         # wait for the worker to have spawned
         with wait_for_call(container, 'spawn_worker'):
-            res = rpc_proxy.service.method.call_async('msg1')
+            res = toxic_rpc_proxy.service.method.call_async('msg1')
 
-        # fake a socket error that causes the proxy to reconnect
-        queue_consumer = rpc_proxy._reply_listener.queue_consumer
-        connection = queue_consumer.consumer.connection
+        try:
+            # disconnect
+            toxiproxy.disable()
 
-        with patch.object(connection, 'drain_events') as drain_events:
-            drain_events.side_effect = ConnectionError
+            # rpc proxy should return an error for the request in flight.
+            # it also attempts to reconnect and throws on failure
+            with pytest.raises(socket.error):
+                with pytest.raises(RpcConnectionError):
+                    res.result()
 
-            # rpc proxy should attempt to reconnect
-            # and return an error for the request in flight
-            with pytest.raises(RpcConnectionError):
-                res.result()
+        finally:
+            # reconnect
+            toxiproxy.enable()
 
-        # proxy should work again afterwards
-        block.send()
-        assert rpc_proxy.service.method("msg3") == "msg3"
+        # proxy will not work afterwards
```

Comment:
we changing the behaviour?

Reply:
No, it's the test.

When we used the management API to close the connection, it closed the existing connection but didn't actually prevent the consumer reconnecting here

https://github.com/nameko/nameko/blob/fb7560e08eaf14fbe1684abf6939efbd7aa9fbca/nameko/standalone/rpc.py#L168-L170

Now, toxiproxy remains disabled until after the consumer fails to reconnect and bubbles out the `socket.error`, so then the following triggers on next use:

https://github.com/nameko/nameko/blob/fb7560e08eaf14fbe1684abf6939efbd7aa9fbca/nameko/standalone/rpc.py#L57-L65

There is a (slightly ugly) hook we can use the re-enable the connection between the first and second attempts, so I will add a test that does that.